src/core/positioning/internalgnssreceiver.cpp

```diff
@@ -103,20 +116,81 @@ void InternalGnssReceiver::handlePositionUpdated( const QGeoPositionInfo &positi
+       || mSatellitesID != satellitesID )
+  {
+    mSatelliteInformationValid = true;
+    mSatellitesUsed = satellitesUsed;
+    mSatellitesID = satellitesID;
+
+    for ( QgsSatelliteInfo &satelliteInfo : mSatellitesInfo )
+    {
+      satelliteInfo.inUse = mSatellitesID.contains( satelliteInfo.id );
+    }
+
+    mLastGnssPositionInformation = GnssPositionInformation( mLastGnssPositionInformation.latitude(),
+                                                            mLastGnssPositionInformation.longitude(),
+                                                            mLastGnssPositionInformation.elevation(),
+                                                            mLastGnssPositionInformation.speed(),
+                                                            mLastGnssPositionInformation.direction(),
+                                                            mSatellitesInfo, 0, 0, 0,
+                                                            mLastGnssPositionInformation.hacc(),
+                                                            mLastGnssPositionInformation.vacc(),
+                                                            mLastGnssPositionInformation.utcDateTime(),
+                                                            QChar(), 0, -1, mSatellitesUsed, QChar( 'A' ), mSatellitesID, mSatelliteInformationValid,
+                                                            mLastGnssPositionInformation.verticalSpeed(),
+                                                            mLastGnssPositionInformation.magneticVariation(),
+                                                            0, mGeoPositionSource->sourceName() );
+    emit lastGnssPositionInformationChanged( mLastGnssPositionInformation );
+  }
+}
+
+void InternalGnssReceiver::handleSatellitesInViewUpdated( const QList<QGeoSatelliteInfo> &satellites )
+{
+  QList<QgsSatelliteInfo> satellitesInfo;
+  for ( const QGeoSatelliteInfo satellite : satellites )
+  {
+    QgsSatelliteInfo satelliteInfo;
+    satelliteInfo.elevation = satellite.hasAttribute( QGeoSatelliteInfo::Elevation )
+                                ? satellite.attribute( QGeoSatelliteInfo::Elevation )
+                                : std::numeric_limits<double>::quiet_NaN();
+    satelliteInfo.azimuth = satellite.hasAttribute( QGeoSatelliteInfo::Azimuth )
+                              ? satellite.attribute( QGeoSatelliteInfo::Azimuth )
+                              : std::numeric_limits<double>::quiet_NaN();
+    satelliteInfo.id = satellite.satelliteIdentifier();
+    satelliteInfo.signal = satellite.signalStrength();
+    satelliteInfo.inUse = mSatellitesID.contains( satellite.satelliteIdentifier() );
+  }
+}
+
 void InternalGnssReceiver::handleError( QGeoPositionInfoSource::Error positioningError )
 {
   qDebug() << positioningError;
   return;
 }
+
+void InternalGnssReceiver::handleSatelliteError( QGeoSatelliteInfoSource::Error satelliteError )
+{
+  qDebug() << satelliteError;
```

Comment:
Reset values / set valid = false?
Especially for [`QGeoSatelliteInfoSource::ClosedError`](https://doc.qt.io/qt-5/qgeosatelliteinfosource.html#Error-enum)

Reply:
@m-kuhn , not a bad idea.